lib/recomb_map.c

```diff
@@ -214,48 +194,39 @@ recomb_map_phys_to_discrete_genetic(recomb_map_t *self, double x, uint32_t *locu
 double
 recomb_map_genetic_to_phys(recomb_map_t *self, double genetic_x)
 {
-    size_t k;
-    double ret = 0.0;
-    double x, s, excess;
-    double *p = self->positions;
-    double *r = self->rates;
-    double num_loci = self->num_loci;
+    int nothing = (int) (self->total_recombination_rate - self->total_recombination_rate); /* Make GCC happy because not using self */
+    return genetic_x + nothing;
+}
 
-    assert(num_loci >= 1);
-    assert(genetic_x >= 0);
-    assert(genetic_x <= num_loci);
-    if (self->total_recombination_rate == 0) {
-        /* When we have a 0 total rate, anything other than m maps to 0. */
-        ret = genetic_x >= self->num_loci? self->sequence_length: 0;
-    } else if (self->size == 2) {
-        /* Avoid roundoff when num_loci == self->sequence_length */
-        ret = genetic_x;
-        if (self->sequence_length != num_loci) {
-            ret = (genetic_x / num_loci) * self->sequence_length;
-        }
-    } else if (genetic_x == num_loci) {
-        /* Map the right end of the chromosome to the right end of the
-         * chromosome, even if there are zero-recomb regions on the end. */
-        ret = self->sequence_length;
-    } else {
-        /* genetic_x is in the range [0,num_loci], and so we rescale
-         * this into [0,total_recombination_rate] so that we can
-         * map back into physical coordinates. */
-        x = (genetic_x / num_loci) * self->total_recombination_rate;
-        if (x > 0) {
-            
-            k = msp_binary_interval_search(x, self->cumulative, self->size);
-            s = self->cumulative[k];
-            assert((s >= x) || (k >= self->size - 1));
-            assert(k > 0);
-            excess = 0;
-            if (r[k - 1] > 0) {
-                excess = (s - x) / r[k - 1];
-            }
-            ret = p[k] - excess;
-        }
-    }
-    return ret;
+double
+recomb_map_position_to_recomb_mass(recomb_map_t *self, double pos)
```

Comment:
Rename to ``recomb_map_position_to_mass()``

A quick comment describing what it does would help too.